src/groupeddataframe/utils.jl

```diff
@@ -346,32 +342,79 @@ function row_group_slots(cols::NTuple{N, AbstractVector},
             end
             refmap
         end
-        @spawn_for_chunks chunk_size for i in eachindex(groups)
-            @inbounds begin
-                seeni = seen_vec[Threads.threadid()]
-                local refs_i
-                let i=i # Workaround for julia#15276
-                    refs_i = map(refarrays) do c
-                        return @inbounds c[i]
+        if use_threading
+            Threads.@threads for i in eachindex(groups)
```

Comment:
`Threads.@threads` won't use multithreading in case of nested parallelism. So `use_threading` should probably be `false` when `Threads.threadid() != 1` to avoid allocating `seen_vec` unnecessarily.

Reply:
OK.